Use an inline namespace for Abseil.

This prevents Drake's build of Abseil from interfering with a user's
downstream build of Abseil when the user is using static linking.

Reasoning for not upstreaming this patch: the options header is specifically
designed for end-user customization, and that's basically all we're doing here.

--- absl/base/options.h
+++ absl/base/options.h
@@ -123,8 +123,8 @@
 // be changed to a new, unique identifier name.  In particular "head" is not
 // allowed.
 
 #define ABSL_OPTION_USE_INLINE_NAMESPACE 1
-#define ABSL_OPTION_INLINE_NAMESPACE_NAME lts_20260107
+#define ABSL_OPTION_INLINE_NAMESPACE_NAME drake_vendor
 
 // ABSL_OPTION_HARDENED
 //

--- absl/debugging/symbolize_elf.inc
+++ absl/debugging/symbolize_elf.inc
@@ -1745,12 +1745,14 @@ bool Symbolize(const void *pc, char *out, int out_size) {
 ABSL_NAMESPACE_END
 }  // namespace absl
 
+#if 0  // This (dead) code disobeys the drake_vendor inline namespace.
 extern "C" bool AbslInternalGetFileMappingHint(const void **start,
                                                const void **end,
                                                uint64_t *offset,
                                                const char **filename) {
   return absl::debugging_internal::GetFileMappingHint(start, end, offset,
                                                       filename);
 }
+#endif

 #undef ABSL_INTERNAL_HAVE_PPC64_ELFV1_ABI
